### PR TITLE
Integrate Consul into gateway microservice

### PR DESCRIPTION
**Pull request by Mewr11**
Update the gateway microservice to use consul load balancing rather than localhost connections.

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -8,13 +8,18 @@ spring:
...
+      port: 8500
+      discovery:
+        prefer-ip-address: true
     
     gateway:
       globalcors:
         corsConfigurations:
           '[/**]':
-            allowedOrigins: "http://localhost:4200"
+            allowedOrigins: "*"
```
Comment (escalonn) [inline src/main/resources/application.yml]: why is this change made? should be answered in the PR description

**Review by escalonn — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml (hunk 1 above)
Comment (Mewr11) [reply src/main/resources/application.yml]: That was a testing change that I thought I reverted.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -8,13 +8,21 @@ spring:
...
     
     gateway:
+      # CORS shouldn't be an issue in the dev/staging environments,
+      # but this is here so a local instance of the frontend can
+      # connect to the dev cluster
       globalcors:
         corsConfigurations:
           '[/**]':
-            allowedOrigins: "http://localhost:4200"
+            allowedOrigins: "localhost:4200"
```
Comment (escalonn) [inline src/main/resources/application.yml]: CORS rules require the origins to include the scheme (http://). if you wanted to include https (not relevant for a local ng serve), that's be a separate origin.

**Review by escalonn — CHANGES_REQUESTED** [review]

**Review by escalonn — CHANGES_REQUESTED** [review]
why isn't the pipeline running?
resolve previous conversations if you've addressed them

**Comment by Mewr11** [comment]
> why isn't the pipeline running?
> resolve previous conversations if you've addressed them

We had an outdated pipeline on this branch.

**Review by escalonn — APPROVED** [review]

**Review by BeoHawk25 — APPROVED** [review]

**Pull request merged**